burr/core/application.py

```diff
@@ -2207,9 +2207,19 @@ def with_state_persister(
         """
         if on_every != "step":
             raise ValueError(f"on_every {on_every} not supported")
+
         if not isinstance(persister, persistence.BaseStateSaver):
             self.lifecycle_adapters.append(persister)
         else:
+            # Check if 'is_initialized' exists and is False; raise RuntimeError, else continue if not implemented
+            try:
```

Comment:
I think this would be cleaner if it has the attribute `is_initialized` (if `hasattr(persister, "is_initialized")`), or something like that...

Reply:
Yes, we're implementing `NotImplemented` to tackle this.